bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/api/MeaterRestAPI.java

```diff
@@ -0,0 +1,157 @@
+
+    public MeaterRestAPI(MeaterBridgeConfiguration configuration, Gson gson, HttpClient httpClient) {
+        this.gson = gson;
+        this.configuration = configuration;
+        this.httpClient = httpClient;
+    }
+
+    public boolean refresh(Map<String, MeaterProbeDTO.Device> meaterProbeThings) {
+        try {
+            // Login
+            login();
+
+            MeaterProbeDTO dto = getDevices(MeaterProbeDTO.class);
+            List<Device> devices = dto.getData().getDevices();
+
+            if (devices != null) {
+                if (!devices.isEmpty()) {
+                    for (Device meaterProbe : devices) {
+                        meaterProbeThings.put(meaterProbe.id, meaterProbe);
+                    }
+                } else {
+                    meaterProbeThings.clear();
+                }
+            }
+            return true;
+        } catch (MeaterException | JsonSyntaxException e) {
+            logger.warn("Failed to refresh! {}", e.getMessage());
+        }
+        return false;
+    }
+
+    private void login() throws MeaterException {
+        try {
+            // Login
+            String json = "{ \"email\": \"" + configuration.email + "\",  \"password\": \"" + configuration.password
+                    + "\" }";
+            Request request = httpClient.newRequest(API_ENDPOINT + LOGIN).method(HttpMethod.POST);
+            request.header(HttpHeader.ACCEPT, JSON_CONTENT_TYPE);
+            request.header(HttpHeader.CONTENT_TYPE, JSON_CONTENT_TYPE);
+            request.content(new StringContentProvider(json), JSON_CONTENT_TYPE);
+
+            logger.debug("HTTP POST Request {}.", request.toString());
+
+            ContentResponse httpResponse = request.send();
+            if (httpResponse.getStatus() != HttpStatus.OK_200) {
+                throw new MeaterException("Failed to login " + httpResponse.getContentAsString());
+            }
+            // Fetch JWT
+            json = httpResponse.getContentAsString();
+            JsonObject jsonObject = JsonParser.parseString(json).getAsJsonObject();
+            JsonObject childObject = jsonObject.getAsJsonObject("data");
+            this.authToken = childObject.get("token").getAsString();
+            this.userId = childObject.get("userId").getAsString();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new MeaterException(e);
+        }
+    }
+
+    private String getFromApi(String uri) throws MeaterException, InterruptedException {
+        try {
+            for (int i = 0; i < MAX_RETRIES; i++) {
+                try {
+                    Request request = httpClient.newRequest(API_ENDPOINT + uri).method(HttpMethod.GET);
+                    request.header(HttpHeader.AUTHORIZATION, "Bearer " + authToken);
+                    request.header(HttpHeader.ACCEPT, JSON_CONTENT_TYPE);
+                    request.header(HttpHeader.CONTENT_TYPE, JSON_CONTENT_TYPE);
```

Comment:
According to the [API documentation](https://github.com/apption-labs/meater-cloud-public-rest-api#api-basics) it is possible to provide an `Accept-Language` header which will then be used to provide localized cook names. This would be simple to support by injecting [LocaleProvider](https://www.openhab.org/javadoc/latest/org/openhab/core/i18n/localeprovider) into the handler from `MeaterHandlerFactory` in order to get the ISO 639-1 language code through `localeProvider.getLocale().getLanguage()`.

Reply:
getLanguage() does not return ISO 639-1, for sweden I get `SE` instead of `sv`.
Maybe it should be a configuration parameter and default to `en`? 

Or can we get ISO 639-1 from core somehow?

Reply:
I'll look into this tomorrow.

Reply:
> ISO3Language: swe

I must have tested wrong method, getLanguage() works as you say. :-)